lambda/translmod.ml

```diff
@@ -692,16 +692,11 @@ let _ =
 
 (* Introduce dependencies on modules referenced only by "external". *)
 
-let scan_used_globals lam =
-  let globals = ref Ident.Set.empty in
-  let rec scan lam =
-    Lambda.iter_head_constructor scan lam;
-    match lam with
-      Lprim ((Pgetglobal id | Psetglobal id), _, _) ->
-        globals := Ident.Set.add id !globals
-    | _ -> ()
-  in
-  scan lam; !globals
+let rec scan_used_globals = function
+  | Lprim ((Pgetglobal id | Psetglobal id), _, _) -> Ident.Set.singleton id
+  | other ->
+      let folder env t = Ident.Set.union (scan_used_globals t) env in
+      fold_left folder Ident.Set.empty other
```

Comment:
Isn't this code quadratic, with every sub-call to `scan_used_globals` traversing not just its subtree, but also all its children?

Reply:
Since `fold_left` here is shallow, I don't think this is the case.

Reply:
Indeed.